docs: Getting Started (#1220)
tendermint/starport: docs/run/index.md

# Start a Blockchain

> Blockchains are decentralized applications that in production run (often) the same software on many validator nodes by different people/entities. To launch a blockchain in production entities coordinate on the launch process together and start their nodes.
> 
> During development, however, a blockchain can be started locally on a single validator node. This is very convenient, because you can quickly restart a chain and iterate faster. Starting a chain on single node in development is similar to starting a traditional web application on a local server.

Reviewer: ```suggestion
Blockchains are decentralized applications (dApps).

- In production, dApps often run the same software on many validator nodes that are run by different people and entities. To launch a blockchain in production, the validator entities coordinate the launch process to start their nodes simultaneously.
- During development, a blockchain can be started locally on a single validator node. This convenient process lets you can restart a chain quickly and iterate faster. Starting a chain on a single node in development is similar to starting a traditional web application on a local server. 
```

Reviewer: hmmm what do we mean here?

> coordinate on the launch process together and start their nodes

I don't know if we really mean simultaneously? (the same time?) 

> the validator entities coordinate the launch process to start their nodes simultaneously.

My main contribution here is to provide parallel sentence structure for the comparison of the production vs development. let's talk it through if needed, we can make this crystal clear.

Author: Removed "dApps", because even though it's short for "decentralized applications", the meaning for me is different. Blockchain is a decentralized application, but dApp is a user-facing app that is powered by one or more chains or contracts. It's confusing, I know.

Author: > I don't know if we really mean simultaneously? (the same time?)

Pretty much at the same time.

## Start a Blockchain Node in Development

> Switch to a directory that contains a blockchain scaffolded with Starport and run the following command:

Reviewer: ```suggestion
Switch to the directory that contains a blockchain that was scaffolded with Starport. To start the blockchain node, run the following command:
```

```
starport serve
```

> This command initializes a chain, builds the code, starts a single validator node and starts watching for file changes. Whenever a file is changed, Starport will reinitialise the chain, rebuild and start the chain again. Starport tries to preserve the state of the chain, if the changes to source code are compatible with the previous state. This is very useful for development purposes. `starport serve` command is a development tool, it should not be used in production environment. See below how to run a chain in production.

Reviewer: ```suggestion
This command initializes a chain, builds the code, starts a single validator node, and starts watching for file changes. 

Whenever a file is changed, the chain is automatically reinitialized, rebuilt, and started again. The chain's state is preserved if the changes to the source code are compatible with the previous state. This state preservation is beneficial for development purposes. 

Because the `starport serve` command is a development tool, it should not be used in a production environment. Read on to learn the process of running a blockchain in production. 
```

Reviewer: I know it's tempting, but let's try not to make Starport an "actor" here. 

Does this rewrite work to prevent anthropomorphism? I like this change for the active voice and clear actions. wdyt?

Author: Sounds good!

## The Magic of `starport serve`

> 

Reviewer: ```suggestion
The `starport serve` command starts a fully operational blockchain.

```

- Creates accounts
- Initializes a blockchain node
- Starts the following processes:
  - Tendermint RPC
  - Cosmos SDK API
  - Faucet, optional
- Watches for file changes and restarts
- Exports state

> The `starport serve` command starts a fully operational blockchain. You can use flags to configure how the blockchain runs. All flags are optional.

Reviewer: ```suggestion
You can use flags to configure how the blockchain runs. 
```

Reviewer: let's move the command description above the bullet list, and let's move "All flags are optional." to the flag description section where it will be easier to consume

## Define How Your Blockchain Starts

> Flags for the `starport serve` command determine how your blockchain starts:

Reviewer: ```suggestion
Flags for the `starport serve` command determine how your blockchain starts. All flags are optional.
```

`--home`

Specify a custom home directory.

## Start a Blockchain Node in Production

> Both `starport serve` and `starport build` compile the source code of the chain in a binary file and install it in `~/go/bin`. By default, the binary name matches the name of the repository with a `d` suffix in the end. For example, if a chain was scaffolded using `starport app github.com/alice/chain`, then the binary will be named `chaind`.

Reviewer: ```suggestion
The `starport serve` and `starport build` commands compile the source code of the chain in a binary file and install the binary in `~/go/bin`. By default, the binary name is the name of the repository appended with `d`. For example, if you scaffold a chain using `starport app github.com/alice/chain`, then the binary is named `chaind`.
```

Reviewer: just a little grammar love

> The binary name can be customised in `config.yml`:

Reviewer: ```suggestion
You can customize the binary name in `config.yml`:
```

Reviewer: let's use US English spelling https://developers.google.com/style/spelling and active voice to match our other content here (I am loving this getting started guide with friendly inclusive language, good job @fadeev !)
